test/xain_test.exs

```diff
@@ -298,4 +298,16 @@ defmodule XainTest do
     end
     assert result == ~s(<div></div>)
   end
+
```

Comment:
Would the expected/safe output of `div "#id.my-class <hello>"` be `"<div class=\"my-class\" id=\"id\">&lt;hello&gt;</div>"`? Because this use case is not yet addressed in this pull request and a test for it would fail. It is not clear to me how i would add the escaping without escaping also generated tags.

Seems that Phoenix.HTML uses a tagged value `{:safe, [...]}` and otherwise escapes strings. In Phoenix you would have to do: `content_tag :div, {:safe, "<hello>"}, id: "id", class: "my-class"` to generate an empty `<hello>`. Treat everything as unsafe, unless tagged as safe seems like a secure design.